tests/utils.cpp

```diff
@@ -75,15 +77,26 @@ TEST(utils, wildcard_match)
   EXPECT_EQ(wildcard_match("foobarbiz", tokens_foo_biz, false, false), true);
 }
 
+static void symlink_test_binary(const std::string& destination)
+{
+  if (symlink("/proc/self/exe", destination.c_str()))
+  {
+    throw std::runtime_error("Couldn't symlink /proc/self/exe to " +
+                             destination + ": " + strerror(errno));
+  }
+}
+
 TEST(utils, resolve_binary_path)
 {
   std::string path = "/tmp/bpftrace-test-utils-XXXXXX";
   if (::mkdtemp(&path[0]) == nullptr) {
     throw std::runtime_error("creating temporary path for tests failed");
   }
+
+  symlink_test_binary(path + "/executable");
```

Comment:
why is the test moved to probe a symlink?

Reply:
We need `path + "/executable"' to represent a real elf file on disk, it's not enough for the file to exist and have the executable bit set. The options I considered is either symlinking or copying and decided to go with symlinking so that size of the binary does not have to be considered. Lmk if I should do something else here.

Reply:
Oh yeah that's probably fine then, makes sense for the unit tests, as the test executable is I think the only test executable you can reliably depend on, maybe just a comment that this is to reference a real ELF file, and that's why you're using bpftrace_test